thirdparty/runtime_environments.go

```diff
@@ -0,0 +1,60 @@
+package thirdparty
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+type RuntimeEnvironmentsClient struct {
+	Client  *http.Client
+	BaseURL string
+	APIKey  string
+}
+
+func NewRuntimeEnvironmentsClient(baseURL string, apiKey string) *RuntimeEnvironmentsClient {
+	c := RuntimeEnvironmentsClient{
+		Client:  &http.Client{},
+		BaseURL: baseURL,
+		APIKey:  apiKey,
+	}
+	return &c
+}
+
+// The function getImageNames returns a list of strings containing the names of all images from the runtime environments API.
+func (c *RuntimeEnvironmentsClient) getImageNames(ctx context.Context) ([]string, error) {
+	apiURL := fmt.Sprintf("%s/rest/api/v1/imageList", c.BaseURL)
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("Api-Key", c.APIKey)
+	resp, err := c.Client.Do(request)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		msg, _ := io.ReadAll(resp.Body)
+		return nil, errors.Errorf("HTTP request returned unexpected status `%s`: %s", resp.Status, string(msg))
+	}
+	var images []string
+	if err := gimlet.GetJSON(resp.Body, &images); err != nil {
+		return nil, errors.Wrap(err, "decoding http body")
+	}
+	defer resp.Body.Close()
+	if images == nil {
```

Comment:
It's probably safer to check for `len(images) == 0`. This will only return if the slice is nil, but if the slice is empty (e.g. because it returned a JSON body of `[]`), then it won't work.

There's [a quick article](https://medium.com/@habibridho/golang-nil-vs-empty-slice-87fd51c0a4d) discussing the difference between nil vs. empty slice.